File: sistema-consultas/src/main/java/br/edu/ifma/consultas/domain/Prescricao.java
package br.edu.ifma.consultas.domain;

public class Prescricao {

    private Long idPrescricao;
    private Medicamento medicamento;
    private String dosagem;
    private String administracao;
    private String tempoDeUso;

    public Prescricao(String administracao, String dosagem, Long idPrescricao, Medicamento medicamento, String tempoDeUso) {
        this.administracao = administracao;
        this.dosagem = dosagem;
        this.idPrescricao = idPrescricao;
        this.medicamento = medicamento;
        this.tempoDeUso = tempoDeUso;
    }

    public String getAdministracao() {
        return administracao;
    }

    public String getDosagem() {
        return dosagem;
    }

    public Long getIdPrescricao() {
        return idPrescricao;
    }

    public Medicamento getMedicamento() {
        return medicamento;
    }

    public String getTempoDeUso() {
        return tempoDeUso;
    }
}

File: sistema_consultas/src/consultas/domain/entities/prescricao.py

from __future__ import annotations
from dataclasses import dataclass
from consultas.domain.value_objects.identificadores import MedicamentoId, PrescricaoId, ProntuarioId

@dataclass(frozen=True, slots=True)
class Prescricao:
    id: PrescricaoId
    prontuario_id: ProntuarioId
    medicamento_id: MedicamentoId
    dosagem: str
    administracao: str
    tempo_de_uso: str
